src/Moryx/Serialization/EntryConvert/EntryConvert.cs

```diff
@@ -749,22 +757,100 @@ public static Entry InvokeMethod(object target, MethodEntry methodEntry)
+        /// Async methods are executed asynchronous
+        /// </summary>
+        public static Task<Entry> InvokeMethodAsync(object target, MethodEntry methodEntry)
+        {
+            return InvokeMethodAsync(target, methodEntry, Serialization);
+        }
+
+        /// <summary>
+        /// Invoke a method on the target object using custom customSerialization
+        /// Async methods are executed asynchronous
+        /// </summary>
+        public static async Task<Entry> InvokeMethodAsync(object target, MethodEntry methodEntry, ICustomSerialization customSerialization)
+        {
+            var (method, arguments) = GetMethodInfoAndArguments(target, methodEntry, customSerialization);
+            object invokeResult;
+
+            // Also part of MethodEntry but be safer than sorry
+            var isAsync = IsAsyncMethod(method);
+            if (isAsync)
+            {
+                var task = (Task)method.Invoke(target, arguments)!;
+                await task.ConfigureAwait(false);
+
+                invokeResult = GetAsyncMethodResult(method, task);
+            }
+            else
+            {
+                invokeResult = method.Invoke(target, arguments);
+            }
+
+            return invokeResult == null ? null : EncodeObject(invokeResult, customSerialization);
+        }
+
+        /// <summary>
+        /// Invoke a method on the target object using custom customSerialization
+        /// Async methods are synchronized
         /// </summary>
         public static Entry InvokeMethod(object target, MethodEntry methodEntry, ICustomSerialization customSerialization)
+        {
+            var (method, arguments) = GetMethodInfoAndArguments(target, methodEntry, customSerialization);
+            object invokeResult;
+
+            var isAwaitable = IsAsyncMethod(method);
+            if (isAwaitable)
+            {
+                var task = (Task)method.Invoke(target, arguments)!;
+                task.GetAwaiter().GetResult();
+
+                invokeResult = GetAsyncMethodResult(method, task);
+            }
+            else
+            {
+                invokeResult = method.Invoke(target, arguments);
+            }
+
+            return invokeResult == null ? null : EncodeObject(invokeResult, customSerialization);
+        }
+
+        private static (MethodInfo, object[]) GetMethodInfoAndArguments(object target, MethodEntry methodEntry, ICustomSerialization customSerialization)
         {
             var method = target.GetType().GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
                 .FirstOrDefault(m => m.Name == methodEntry.Name && (m.IsPublic || m.IsAssembly) && ParametersProvided(m.GetParameters(), methodEntry));
             if (method == null)
             {
-                throw new MissingMethodException();
+                throw new MissingMethodException($"Method {methodEntry.Name} not found on target!");
```

Comment:
Want to add the target type to the exception for an even better error experienca?